mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -42,7 +42,7 @@ void createGemmTuningRangeBF(struct TunableParams *newSpace,
       {8, 16, 32},
       {4, 8, 16, 32, 64, 128},
       {4, 8, 16, 32, 64, 128},
-      {1, 4, 8},
+      {1, 4, 8, 16},
```

Comment:
I'm giving you a heads up that too wide kPack might cause issues in conv side because we may run out of registers. So I think you may want to wait for my PR to merge and then run this through convolution tuning of the CI to make sure not regression will show up.

Reply:
This is only the tuning though. I mean, if in convolution 16 is too big, KPack 4 or 8 will be selected. I am only telling to the tuner to try a kPack a bit bigger for int8. Actually, I also see that the search space is similar for int8 and floating points. Should we unify that as well?

Reply:
> I am only suggesting to the tuner to try an additional kPack for int8. 

If it is only for our end, then I'm good. However, if there's a perf benefit from the extended kPack range, it'd be bad if MIOpen tuning can't take advantage of it. The thing in MIOpen end is, we have a requirement that once the applicability check pass, kernel run shouldn't fail.

This is the kind of case where the applicability check will indeed pass because there's no way to tell if registers are used up. However, when it fail, it fail in generation of the kernel.

>  I also see that the search space is similar for int8 and floating points. Should we unify that as well?

You can give it a shot but I suspect it will run into exact issue as I mentioned above. A longer K dimension might cause non-int8 type to file after we report ourselves to be applicable, it will end to be an exception on MIOpen end. That's the main reason I want the change to be tested through MIOpen.

Reply:
Can we have a chat about this? I am still unclear how MIOpen uses our things and this might be a good occasion for me to learn that

Reply:
Sure - let's bring Jerry in on it though, since he can find the MIOpen code faster than I can

Reply:
(... alternative thought, we - and MIOpen - may want to limit the kPack value to what's applicable for a given datatype)